### PR TITLE
Delete review from admin dashboard

### PR DESCRIPTION
**Pull request by hyunzeekim**
## Notion ticket link

[Delete Review from Admin Dashboard](https://www.notion.so/uwblueprintexecs/Delete-Review-from-Admin-Dashboard-3ff511af5a7246d09b0b38cc48861ea0)

## Implementation description
* When delete icon in a row is clicked, open confirmation dialog
* Clicking delete review removes the review from data and removes the row
* Clicking go back does not delete the review

## Steps to test
1. Delete review, and verify that the row disappears
2. Refresh the page to verify that the removed review is deleted
3. Go back on confirmation dialog and verify that the row does not disappear
4. Refresh the page to verify that the review was not removed

## What should reviewers focus on?
* code & correctness

## Checklist
- [X] My PR name is descriptive and in imperative tense
- [X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X] I have run the appropriate linter(s)
- [X] I have run docker-compose up and my project compiled
- [X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx
Comment (susheelk) [inline frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx:43]: Could you make this name more descriptive? ie `isDeleteModalOpen`

File: frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx
Comment (susheelk) [inline frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx:54]: (minor) Maybe we can combine these into 1 line, ie `const newData = [...newData.splice(reviewIndex, 1)];`

Comment (susheelk) [inline frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx:54]: (minor) Maybe we can combine these into 1 line, ie `const newData = [...newData.splice(reviewIndex, 1)];`

File: frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx
Comment (susheelk) [inline frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx:185]: I think it's a good idea to put this modal in a separate file to avoid clutter

Comment (susheelk) [inline frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx:185]: I think it's a good idea to put this modal in a separate file to avoid clutter

**Review by susheelk — CHANGES_REQUESTED** [review]
Good job! Added a few comments

File: frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx
Comment (hyunzeekim) [reply frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx:54]: Hmm not sure that would work.. I get a weird behaviour where only the deleted review remains until the page is refreshed.

Comment (susheelk) [reply frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx:51]: Alright nws lol it was nitpick

File: frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx
Comment (dhmi19) [inline frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx:164]: instead of disabling the lint, can you add the types to these variables? 
image

File: frontend/src/components/pages/AdminDashboard/DeleteConfirmation.tsx
Comment (dhmi19) [inline frontend/src/components/pages/AdminDashboard/DeleteConfirmation.tsx:20]: Can we call this DeleteConfirmationModal instead?

Comment (susheelk) [reply frontend/src/components/pages/AdminDashboard/DeleteConfirmation.tsx:20]: +1

File: frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx
Comment (susheelk) [inline frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx:38]: Could you make these names more descriptive as well? ie explicitly state that this is the review to be deleted

File: frontend/src/components/pages/AdminDashboard/DeleteConfirmation.tsx
Comment (susheelk) [inline frontend/src/components/pages/AdminDashboard/DeleteConfirmation.tsx:20]: Could you add a docstring describing this component?

**Review by susheelk — CHANGES_REQUESTED** [review]

File: frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx
Comment (hyunzeekim) [reply frontend/src/components/pages/AdminDashboard/AdminDashboard.tsx:164]: The lint disabling was because the `value` and `updateValue` are unused variables but I need to include them to not get an error when using `tableMeta` for some reason.

**Comment by susheelk** [comment]
Tested and functionality looks good! :partying_face: 
### Cases
* Clicked the delete button, and pressed "take me back": didn't delete review :heavy_check_mark: 
* Clicked the delete button and confirmed:
  * Review removed from mui table :heavy_check_mark: 
  * Review removed from reviews table :heavy_check_mark: 
  * All associated books removed :heavy_check_mark: 

Good to merge pending conflict resolution and @dhmi19 's approval!

**Review by susheelk — APPROVED** [review]
All good! Nice work :balloon:

**Pull request merged**
